src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -7821,4 +7821,7 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_ExpectedInterpolatedString" xml:space="preserve">
     <value>Expected interpolated string</value>
   </data>
+  <data name="ERR_CollectionExpressionImmutableArray" xml:space="preserve">
+    <value>'{0}' requires a CollectionBuilderAttribute to be used with collection expressions.</value>
```

Comment:
I'm actually wondering if even this language is a bit too prescriptive; almost nobody who receives this message will have control over the ImmutableArray type, and if they do anything to make the scenario work, it will probably be to define ImmutableCollectionsMarshal.

We could consider saying something like `This version of '{0}' cannot be used with collection expressions`.

I don't feel super strongly about it and I'm curious what others think.

Reply:
I think the reword makes sense; I, as a user, can't do anything to address this message. The package reference needs to be updated.